### PR TITLE
Use `pin-project-lite` instead of `pin-project`

### PR DESCRIPTION
**Pull request by sunli829**
The compilation speed has not improved much, at least it looks like this on my Mac. But pin_project_lite is not a procedural macro, so the IDE can correctly analyze the return type of the `project` function. I believe this can make the code easier to maintain.

### CODE REVIEW

**Comment by davidpdrsn** [comment]
Thanks for doing this! Using `pin-project-lite` totally makes sense.

As discussed [here](https://github.com/tokio-rs/axum/pull/94) though I think we should keep the import ordering as it was. That will also make this change smaller and easier to review.

**Comment by sunli829** [comment]
> As discussed [here](https://github.com/tokio-rs/axum/pull/94) though I think we should keep the import ordering as it was. 

Yes, I am working on this and it will be done soon. 😁

**Comment by davidpdrsn** [comment]
Thank you 😃

**Comment by sunli829** [comment]
I have done it. 😁

File: src/ws/mod.rs [hunk 1]
```diff
@@ -212,6 +216,47 @@ impl<F, B, T> WebSocketUpgrade<F, B, T> {
...
+        I: IntoIterator,
+        I::Item: Into<Cow<'static, str>>,
+    {
+        self.protocols = protocols
+            .into_iter()
+            .map(Into::into)
+            .collect::<Vec<_>>()
+            .into();
+        self
+    }
```
Comment (davidpdrsn) [inline src/ws/mod.rs:259]: Seems you need to rebase/merge main as this was merged in the other PR.

**Comment by sunli829** [comment]
Sorry, it's done now. 😂

**Review by davidpdrsn — APPROVED** [review]
Thanks!

**Pull request merged**